File: ps/sub/subscriber.py
# -*- coding: utf-8 -*-

import sys
import zmq

port = "5555"
if len(sys.argv) > 1:
  port =  sys.argv[1]
  int(port)

context = zmq.Context()
socket = context.socket(zmq.SUB)

print("Connecting to server…")
socket.connect ("tcp://localhost:%s" % port)

topic = "pubsub"
socket.setsockopt_string(zmq.SUBSCRIBE, topic)

while True:
  topic = socket.recv()
  message = socket.recv()
  
  print("Received from channel %s:", topic)
  print(message)

File: ps/sub/subscriber.js
const zmq = require("zeromq");

async function runClient() {
  console.log("Connecting to server…");

  const sock = new zmq.Subscriber();
  sock.connect("tcp://localhost:5555");
  sock.subscribe("pubsub");

  for await (const [topic, msg] of sock) {
    console.log("Received from channel %s:", topic);
    console.log(JSON.parse(msg));
  }
}

runClient();
